assets/environment-interpreter/activate/activate

```diff
@@ -166,6 +170,16 @@ while true; do
       _FLOX_ENV_DESCRIPTION="$1"
       shift
       ;;
+     --keep-env-vars)
```

Comment:
I'm unsure about adding more knobs and increasing the feature surface area of activation for a build-specific feature. Could this still be done in `flox-build.mk` instead?

Reply:
No, not without putting a messy `sh -c 'env -i $(foreach i,$(ALLOW_OUTER_ENV_VARS),$(i)="$$$$$(i)") flox activate ...'` invocation in-between the outer and inner invocations for the purpose of deferring the evaluation of variables like `$CPATH`. Doing it this way is simpler and easier to debug. And if that flag is only used for builds then it has zero chance of affecting anything else so I'm confident it's safe as well.

Reply:
Note that this change was removed from the activate script and instead added to the wrapper script, so is now specific to builds only.

Reply:
Note that this change was removed from the activate script and instead added to the wrapper script, so is now specific to builds only.